Gems/OpenXRVk/3rdParty/Platform/Android/FindOpenXROculus.cmake

```
@@ -0,0 +1,42 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+# 
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+# this file actually ingests the library and defines targets.
+set(TARGET_WITH_NAMESPACE "3rdParty::OpenXROculus")
+if (TARGET ${TARGET_WITH_NAMESPACE})
+    return()
+endif()
+
+set(MY_NAME "OpenXROculus")
+
+get_property(openxrvk_gem_root GLOBAL PROPERTY "@GEMROOT:OpenXRVk@")
+
+set(OculusOpenXRSDKPath ${openxrvk_gem_root}/External/OculusOpenXRMobileSDK)
+
+set(${MY_NAME}_INCLUDE_DIR 
+    ${OculusOpenXRSDKPath}/3rdParty/khronos/openxr/OpenXR-SDK/include
+    ${OculusOpenXRSDKPath}/OpenXR/Include)
+
+set(PATH_TO_SHARED_LIBS ${OculusOpenXRSDKPath}/OpenXR/Libs/Android/arm64-v8a)
+
+if(NOT EXISTS ${PATH_TO_SHARED_LIBS}/Release/libopenxr_loader.so)
+    message(FATAL_ERROR
+        "Oculus OpenXR loader library not found at ${PATH_TO_SHARED_LIBS}/Release. "
+        "Oculus OpenXR Mobile SDK needs to be downloaded via https://developer.oculus.com/downloads/native-android/ "
+        "and uncompressed into OpenXRVk/External/OculusOpenXRMobileSDK folder.")
+    return()
+endif()
+
+set(${MY_NAME}_SHARED_LIBRARIES ${PATH_TO_SHARED_LIBS}/$<IF:$<CONFIG:debug>,Debug,Release>/libopenxr_loader.so)
+
+add_library(${TARGET_WITH_NAMESPACE} INTERFACE IMPORTED GLOBAL)
```

This should actually set the library type to SHARED IMPORTED GLOBAL and the set the INTERFACE_LOCATION to the libopenxr_loader.so path.
We had issues with this in the past with libtiff and zlib not causing linker errors because the order in which the libraries are specified on the command line are unspecified, because `target_link_libraries` was used.

This allows `3rdParty::OpenXROculus` target to properly be supplied to the correct order to the linker if another library depends on it.
```suggestion
set(${MY_NAME}_SHARED_LIBRARIES ${PATH_TO_SHARED_LIBS}/$<IF:$<CONFIG:debug>,Debug,Release>/libopenxr_loader.so)

add_library(${TARGET_WITH_NAMESPACE} SHARED IMPORTED GLOBAL)
```

---

I'll test the suggestion

---

One question, `libopenxr_loader.so` is loaded manually by code, it's not suppose to link against it. Using `SHARED` will not link against the .so, will it?

---

Sorry, scratch that. It's the vulkan library what the loader loads dynamically, not the loader. It'd be nice for this loader to be dynamic in the future too, so one executable can handle both oculus loader and official loader, but that's something we will worry about later.

---

Change done and tested.